tests/common.py

```diff
@@ -101,13 +101,12 @@ def tensors_default_to(host):
     :param str host: Either "cuda" or "cpu".
     """
     assert host in ("cpu", "cuda"), host
-    old_module, name = torch.Tensor().type().rsplit(".", 1)
-    new_module = "torch.cuda" if host == "cuda" else "torch"
-    torch.set_default_tensor_type("{}.{}".format(new_module, name))
+    old_host = torch.Tensor().device
+    torch.set_default_device(host)
```

Comment:
Do we need this context manager at all? How is it different from `with torch.device(device)`? https://pytorch.org/docs/stable/generated/torch.set_default_device.html

> To only temporarily change the default device instead of setting it globally, use with torch.device(device): instead.

Reply:
Oh it would be great to replace this custom context manager with `torch.device`, as long as `torch.device` exists in our earliest supported torch version 1.11.0

Reply:
It looks like `torch.set_default_device` is new in PyTorch 1.12

https://github.com/pytorch/pytorch/releases

https://docs.google.com/spreadsheets/d/1H3jazwO8BBCwK8JwLNYspLiHfUrzshEtyqjL-X93I9g/edit#gid=790902532

Reply:
ok, well let's keep the polyfill until we drop support for torch==1.11.

Reply:
Should we drop 1.12 support? Otherwise the code will get ugly I think

Reply:
To clarify: this also applies to `torch.set_default_device` throughout this PR and we if we want to keep torch 1.11 support then there will be a lot of if/else statements between `torch.set_default_device` and `torch.set_default_tensor_type`

Reply:
Hmm.. it looks like `set_default_device` is not exposed or not available even in torch 1.13

https://pytorch.org/docs/1.13/search.html?q=set_default&check_keywords=yes&area=default

Reply:
Pyro has always aimed at being more stable than torch, and we have historically implemented [polyfills](https://en.wikipedia.org/wiki/Polyfill_(programming)) in Pyro to smooth over PyTorch's move-fast-and-break-things attitude.  If I had time, I'd implement polyfills like a `pyro.util.device()` context manager, a `pyro.set_default_device()` helper. etc.  But I don't have time, and maybe it's time to drop this aim as our maintenance resources dwindle.  🤷 

The motivation for being very stable is to avoid giving people headaches.  Every time we drop support for some version of some underlying library, some grad student wastes a day trying to install an old repo whose author has graduated and didn't pin versions.  Every time we pin a library version, some software engineer at BigCo wastes a week solving dependency issues between conflicting libraries with non-overlapping version pins, spending a day committing to an upstream repo maintained by an overcommitted professor, building polyfills around another dependency (who doesn't explicitly pin versions but actually depends on a version outside our range, which it took haf a day to figure out), and replacing a third library that is no longer maintained.

If you do decide to drop torch 1.11 support, could you update the version pins everywhere and update the python supported versions?  And we'll bump the minor version in our next release.

Reply:
> Hmm.. it looks like `set_default_device` is not exposed or not available even in torch 1.13
> 
> https://pytorch.org/docs/1.13/search.html?q=set_default&check_keywords=yes&area=default

I confirmed this by installing it locally:

```
>>> import torch
>>> torch.__version__
'1.13.0+cu117'
>>> torch.set_default_device
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'torch' has no attribute 'set_default_device'
```

Can you confirm that it's ok to drop the support for all torch 1.11, 1.12, and 1.13?

Reply:
Sure, let's just be sure to announce in our release notes and bump the minor version.